api_app/core/serializers.py

```diff
@@ -174,6 +200,8 @@ def read_and_verify_config(cls) -> dict:
         This function is memoized for the md5sum of the JSON file.
         """
         config_dict = cls._read_config()
+        for plugin in config_dict:
+            cls._complete_config(config_dict, plugin, set())
```

Comment:
is this loop needed here? I think you could just call the function with `config_dict` as a single param and then do the loop inside the function

Reply:
Then I would have to create another function as the loop itself needs to call a function. For reference, this is just like Depth-First Traversal of a directed graph. The loop ensures that all components of the graph are traversed.

Reply:
thanks for clarification, that makes sense